src/Symfony/Component/PropertyInfo/PropertyTypeExtractorInterface.php

```diff
@@ -11,17 +11,24 @@
 
 namespace Symfony\Component\PropertyInfo;
 
+use Symfony\Component\PropertyInfo\Type as LegacyType;
+use Symfony\Component\TypeInfo\Type;
+
 /**
  * Type Extractor Interface.
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @method Type|null getType(string $class, string $property, array $context = [])
  */
 interface PropertyTypeExtractorInterface
 {
     /**
      * Gets types of a property.
      *
-     * @return Type[]|null
+     * @deprecated since Symfony 7.1, use "getType" instead.
```

Comment:
I'm not sure that it'll help here.
Instead of doing `method_exists` stuff, we'll do `instanceof` stuff, but the rest will remain the same.
Plus, creating a new interface will require finding a new name (and the current one fits perfectly).

But maybe I'm missing something?

Reply:
looking at it, I don't think an interface would make it simpler either